src/main/resources/com/openshift/jenkins/plugins/OpenShiftDSL.groovy

```diff
@@ -905,6 +909,38 @@ class OpenShiftDSL implements Serializable {
         return newObjectsAction("startBuild", "start-build", args);
     }
 
+    public boolean verifyService(String svcName) {
+        OpenShiftResourceSelector svcSel = this.selector("svc", svcName);
+        String ip = (String)svcSel.object().get("spec").get("clusterIP");
+        String port = (String)svcSel.object().get("spec").get("ports").get(0).get("port");
+        int i = 0;
+        while (i < 5) {
+            i++;
+            InetSocketAddress address = new InetSocketAddress(ip, Integer.parseInt(port));
+            Socket socket = null;
+            try {
+                socket = new Socket();
+                socket.connect(address, 2500);
```

Comment:
current maximum execution time is 12.5 seconds - is this too long?

Reply:
it was what we did with the old plugin

and users can override with the `timeout(...)` pipeline step enclosing a call to our new method here

Reply:
catching the interrupt exception makes more sense with that timeout